lib/validates_zipcode/formatter.rb

```diff
@@ -0,0 +1,50 @@
+module ValidatesZipcode
+  module Formatter
+    class << self
+      # we assume the zipcode is valid for that country
+      def format(zipcode, country_alpha2)
+        method = "format_#{country_alpha2.to_s.downcase}"
+        respond_to?(method) ? send(method, zipcode) : format_default(zipcode)
+      end
+
+      def format_default(zipcode)
+         zipcode.to_s.strip
+      end
+
+      def format_at(zipcode)
+        zipcode.to_s.scan(/\d/).join
+      end
+
+      def format_de(zipcode)
+        digits = zipcode.to_s.scan(/\d/)
+        # fill leading zeros
+        (5 - digits.count).times { digits.insert(0, '0') }
+        digits.join
```

Comment:
I think this can be simplified as:

```suggestion
       zipcode.to_s.rjust(5, "0")
```

Reply:
awesome hint! Thanks!